### ISSUE TITLE
implement cross-validation

### ISSUE THREAD

**Issue by JakeTheWise** [keras-team/keras-tuner#139]
Would be nice to extend the idea of `executions_per_trial` to CV folds, reporting the average eval metric to the tuner.

**Comment by omalleyt12** [comment]
@JakeTheWise Thanks for the issue! 

Agreed. [This issue](https://github.com/keras-team/keras-tuner/issues/122) describes some of the challenges involved in providing built-in cross-validation for Keras models given the wide range of data that Keras accepts, and also gives an example of how you could override Tuner to support this.

I think with upcoming versions we will try to figure out a way to support cross-validation in a standardized way, which will be a collaborative effort b/t keras-tuner, tf.keras, and tf.data

**Comment by omalleyt12** [comment]
My current thinking is we should only support cross-validation for data that `sklearn` Cross-validation supports, and leverage those utilities. We'd create a `Tuner` subclass `CrossValidationTuner`

**Comment by peacej** [comment]
This sounds very useful, even if it's only available for certain types of data formats. Looking forward to it!

**Comment by voquangtuong** [comment]
Hello omalleyt12. Could you explain more about how to create subclass CrossValidationTuner. We are very interested in it. Thank you
